modules/common/condition/funcs.go

```diff
@@ -380,6 +380,17 @@ func (conditions *Conditions) Mirror(t Type) *Condition {
 	return mirrorCondition
 }
 
+// RestoreLastTransitionTimes - Updates each condition's LastTransitionTime when its state
+// matches the one in a list of "saved" conditions.
+func RestoreLastTransitionTimes(conditions *Conditions, savedConditions Conditions) {
+	for idx, cond := range *conditions {
+		savedCond := savedConditions.Get(cond.Type)
+		if savedCond != nil && HasSameState(&cond, savedCond) {
```

Comment:
OK, so we went with the more fine grained transition detection. I.e. if only the message changes we bump the transition time. That is OK to me.